src/modules/crud.js

```diff
@@ -0,0 +1,86 @@
+// Constants:
+const addTask = document.getElementById('input');
+const list = document.getElementById('tasks-list');
+
+function crud() {
+  list.innerHTML = '';
+  const tasksList = localStorage.getItem('data')
+    ? JSON.parse(localStorage.getItem('data'))
+    : [];
+  for (let i = 0; i < tasksList.length; i += 1) {
+    const input = document.createElement('input');
+    input.setAttribute('type', 'checkbox');
+    input.classList.add('checkbox');
+    input.id = i + 1;
+    if (tasksList[i].completed) {
+      input.checked = true;
+    }
+
+    const label = document.createElement('input');
+    label.value = `${tasksList[i].description}`;
+    label.classList.add('description');
+    label.setAttribute('type', 'text');
+    if (tasksList[i].completed) {
+      label.style.textDecorationLine = 'line-through';
+    }
+
+    label.setAttribute('readOnly', 'readOnly');
+    label.classList.add('description');
+
+    const edit = document.createElement('button');
+    edit.id = i + 1;
+    edit.innerHTML = '✏️';
+    edit.classList.add('edit-btn');
+
+    const li = document.createElement('li');
+    li.classList.add('container-li');
+    li.id = i + 1;
+
+    li.appendChild(input);
+    li.appendChild(label);
+    li.appendChild(edit);
+    list.appendChild(li);
+
+    addTask.value = '';
+
+    input.addEventListener('input', (event) => {
+      if (input.checked) {
+        const idChecked = event.target.id;
+        const data = JSON.parse(localStorage.getItem('data'));
+        data[idChecked - 1].completed = true;
+        label.style.textDecorationLine = 'line-through';
+        localStorage.setItem('data', JSON.stringify(data));
+      } else if (!input.checked) {
+        const idNoChecked = event.target.id;
+        const data = JSON.parse(localStorage.getItem('data'));
+        data[idNoChecked - 1].completed = false;
+        label.style.textDecorationLine = 'none';
+        localStorage.setItem('data', JSON.stringify(data));
+      }
+      window.location.reload();
+    });
+
+    edit.addEventListener('click', (event) => {
+      if (edit.innerHTML === '✏️') {
+        label.removeAttribute('readonly');
+        label.focus();
+        edit.innerHTML = '💾';
+      } else {
+        const edited = label.value;
+        const idNumber = event.target.id;
+        label.setAttribute('readonly', 'readonly');
+        edit.innerHTML = '✏️';
+
```

Comment:
- [ ] Good job so far 👏 You are almost there 😃. You have implemented almost all functionalities except the delete task functionality. **I can not be able to delete a task.** I have seen that you implemented the functionality to clear or delete all tasks that are completed. Well, **that's great but we have to implement the feature that allows the user to delete or remove a task whenever he/she needs regardless of whether it is completed or not** 😃 so please implement that 🏃 
**NOTE** Remember to keep the remaining tasks' indexes implemented. This is among the project requirements too and kudos for you have implemented it on the clearAllComplete feature 😉

Reply:
Made the requested changes, and now I have a button to delete any task.